config/routes.rb

```diff
@@ -401,6 +401,15 @@
           put 'test_connection'
         end
         resources :images, :except => [:show]
+        resources :key_pairs, :only => [:index] do
+          member do
+            get 'download'
+            put 'recreate'
+          end
+          collection do
+            delete 'remove'
```

Comment:
because I am not deleting an AR object with id. I am deleting a Fog::Compute::AWS::KeyPair object which has a name. The name is passed as a parameter. 
hmm... I guess I can pass the name param as id